polaris-core/src/main/java/io/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java

```diff
@@ -143,14 +143,35 @@ public void setStorageIntegrationProvider(PolarisStorageIntegrationProvider stor
    */
   private PolarisMetaStoreManager.PrincipalSecretsResult
       bootstrapServiceAndCreatePolarisPrincipalForRealm(
-          RealmContext realmContext, PolarisMetaStoreManager metaStoreManager) {
+          RealmContext realmContext,
+          PolarisMetaStoreManager metaStoreManager,
+          boolean overwrite) {
     // While bootstrapping we need to act as a fake privileged context since the real
     // CallContext hasn't even been resolved yet.
     PolarisCallContext polarisContext =
         new PolarisCallContext(
             sessionSupplierMap.get(realmContext.getRealmIdentifier()).get(), diagServices);
     CallContext.setCurrentContext(CallContext.of(realmContext, polarisContext));
 
+    if (!overwrite) {
+      PolarisMetaStoreManager.EntityResult preliminaryRootPrincipalLookup =
+          metaStoreManager.readEntityByName(
```

Comment:
Will this be version safe in the future? For now I assume this is safe but in the future could readEntityByName fail for a Polaris catalog bootstrapped from an earlier version?

This seems like a good check to have for now, I'm just more worried about what happens if this call fails because the current version of Polaris just doesn't understand how to look up the Principal in the past.

Reply:
This is an important concern and it's why we don't rely on a check like this to auto-bootstrap... but it seems to me that this check is much less critical. That scenario would be very bad for other reasons, but here you would only be losing the protection of the `--overwrite` flag.